docs/gcp_kms.md

```diff
@@ -0,0 +1,190 @@
+Project name is required in the signatory config.
+
+###
+**Basic permissions**
+* `cloudkms.cryptoKeyVersions.get`
+* `cloudkms.cryptoKeyVersions.list`
+* `cloudkms.cryptoKeyVersions.viewPublicKey`
+* `cloudkms.cryptoKeys.get`
+* `cloudkms.cryptoKeys.list`
+
+###
+**Sign**
+* `cloudkms.cryptoKeyVersions.useToSign`
+###
+**Import**
+
+* `cloudkms.cryptoKeyVersions.create`
+* `cloudkms.cryptoKeys.create`
+* `cloudkms.importJobs.create`
+* `cloudkms.importJobs.get`
+* `cloudkms.importJobs.list`
+* `cloudkms.importJobs.useToImport`
+
+###
+**Configuration parameters \
+Below are the configuration fields which are required for Signatory.**
+
+<table>
+  <tr>
+   <td>
+<strong>Name</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Required</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>application_credentials
+   </td>
+   <td>string
+   </td>
+   <td>
+   </td>
+   <td>Path to the GCP application token JSON file (overrides <code>GOOGLE_APPLICATION_CREDENTIALS</code> environment variable)
+   </td>
+  </tr>
+  <tr>
+   <td>application_credentials_data
+   </td>
+   <td>string
+   </td>
+   <td>
+   </td>
+   <td>GCP application token JSON data (overrides <code>application_credentials</code>)
+   </td>
+  </tr>
+  <tr>
+   <td>project
+   </td>
+   <td>string
+   </td>
+   <td>✅
+   </td>
+   <td>Project name
+   </td>
+  </tr>
+  <tr>
+   <td>location
+   </td>
+   <td>string
+   </td>
+   <td>✅
+   </td>
+   <td>Location
+   </td>
+  </tr>
+  <tr>
+   <td>key_ring
+   </td>
+   <td>string
+   </td>
+   <td>✅
+   </td>
+   <td>Key ring name
+   </td>
+  </tr>
+</table>
+
+###
+
+**Key Management**
+
+Under <code>key management</code> create a new <code>key-ring</code> with any location and create a key with <code>purpose</code> as<strong> <code>Asymmetric sign </code></strong>and<code> protection level </code>as<code>HSM.</code>
+
+The key-ring name and location are required in the signatory configuration.
+
+###
+
+**Application Access:**
+
+The below steps are for providing signatory with the permissions to access the google cloud account Key Management.
+
+* Select `IAM & ADMIN` from the menu and select `Service accounts`. Create a new service account or use an existing one with all the above permissions (Get, Sign & Import) granted.
+* Select the created/existing service account and within that create a new key and a``prompt to download the application credentials will appear, select the JSON format.
+* The downloaded JSON file is needed in signatory config or can be assigned to the below environment variable.
+
+###
+
+**Environment variables**
+
+`cloudkms` backend accepts GCP's standard `GOOGLE_APPLICATION_CREDENTIALS` environment variable
+
+```
+export GOOGLE_APPLICATION_CREDENTIALS="signatory-testing-a7sdfew625aecb.json"
+```
+
+**Getting PKH**
+
+```
+abineshm@Abineshs-MacBook-Pro signatory % ./signatory-cli list -c /etc/s.yaml
```

Comment:
I see other places with the CLI commands just starting with %, can you remove the personal info from this snippet and the next?